Skynet/src/main/java/Faction.java

```diff
@@ -0,0 +1,52 @@
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class Faction implements Runnable {
+
+    @Getter
+    private final String name;
+    private final Factory factory;
+
+    private final Queue<RobotPart> inventory = new ConcurrentLinkedQueue<>();
+
+
+    @Override
+    public void run() {
+        for (int day = 1; day <= SimulationConfig.DAYS; day++) {
+            sleepRandomInterval();
+
+            List<RobotPart> collected = factory.collectParts(SimulationConfig.FACTION_PARTS_PER_DAY);
+            inventory.addAll(collected);
+
+            logCollection(day, collected.size());
+        }
+    }
+
+    public int getCompleteRobotCount() {
+        Map<PartType, Long> counts = inventory.stream()
+                .collect(Collectors.groupingBy(RobotPart::type, Collectors.counting()));
+
+        return Arrays.stream(PartType.values())
+                .mapToInt(type -> counts.getOrDefault(type, 0L).intValue())
+                .min()
+                .orElse(0);
+    }
+
+    private void sleepRandomInterval() {
```

Comment:
sleepRandomInterval() Call: The sleep call at the beginning of the run() loop is meant to simulate the factions "going to the Factory at night," but because the factory is producing parts concurrently, this sleep time doesn't correctly align the actions. It just adds a random delay before they try to collect parts from the factory's currently available inventory. The sleep should be a part of the synchronized, daily cycle, not an arbitrary delay.